### PR TITLE
refactor: error pages

### PR DESCRIPTION
**Pull request by 2fd**
This PR changes all error pages:

- normalize all the styles
- makes them mobile friendly
- prevents error message to go out of the screen

## mobile no supported

Screen Shot 2022-05-26 at 10 56 49

## fatal errors

Screen Shot 2022-05-26 at 10 58 14

Screen Shot 2022-05-26 at 10 57 57

## fatal errors with big messages

Screen Shot 2022-05-26 at 10 58 40

Screen Shot 2022-05-26 at 10 58 25

 ## network mismatch

Screen Shot 2022-05-26 at 10 58 54

## another session detected

Screen Shot 2022-05-26 at 10 59 11

## error connecting with catalyst

Screen Shot 2022-05-26 at 10 59 20

### storybook

Download the branch and run `npm run storybook` you can test each error individually

Screen Shot 2022-05-26 at 11 15 15

### live

you can reproduce each error individually:

- **mobile not supported**: open it on your mobile phone
- **network mismatch**: change your wallet network
- **device not supported**: open it on your commodore 64
- **another session detected**: open it on two different tabs/browsers at the same time
- **not connecting with peers**: ?
- **fatal error**: ?

### CODE REVIEW

**Comment by menduz** [comment]
awesome

File: src/components/errors/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  </div>
+})
+
+export const Image = React.memo(function ({ className, ...props }: React.ImgHTMLAttributes<HTMLImageElement>) {
+  return <div className={classJoin(['error-image', className])}>
+    <img alt="error" {...props} />
+  </div>
+})
+
+export function reload() {
```
[the remark this answers is not in the document]
Comment (2fd) [reply src/components/errors/Error.tsx:33]: no, just on error

File: src/utils/classJoin.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function classJoin(items: (string | false | undefined | null)[]): string {
```
[the remark this answers is not in the document]
Comment (2fd) [reply src/utils/classJoin.ts]: ok

File: src/integration/mobile.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react"
+
+export function useMobileResize(active: boolean = true) {
+  const [ size, setSize ] = useState(() => ({ height: 0, width: 0 }))
+  useEffect(() => {
+    if (!active) {
+      return
+    }
+
+    const root = document.getElementById('root')
```
[the remark this answers is not in the document]
Comment (2fd) [reply src/integration/mobile.ts:10]: this is just a refactor, it doesn't create new code, we can move it into the context in the feature, passing the id as parameter doesn't solve the problem cause you still need the id that is outside the React lifecycle

File: src/integration/mobile.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    if (root) {
+      root.className += ' full'
```
[the remark this answers is not in the document]
Comment (2fd) [reply src/integration/mobile.ts:12]: yes, we need to apply a global style, that is why we are applying the change. In any case this is just a refactor I just move code that is already in production: https://github.com/decentraland/explorer-website/blob/c8930fc7e1922f2a13caf22cbd961940e5176941/src/components/common/StreamContainer.tsx#L29-L46

we can change it later, it that ok for you?

**Comment by 2fd** [comment]
Screen Shot 2022-05-26 at 20 39 26

It is ok, it just showing what is define in code

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**